### PR TITLE
Update ServletMapping for Async dispatch

### PR DESCRIPTION
**Pull request by pmd1nh**
https://github.com/openliberty/open-liberty/issues/22971

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build
       #spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
Added traces to Servlet 4.0 HttpServletMappingImpl.
Updated servlet.4.0_fat servletMapping async tests to verify the new behaviors when repeat in EE10.

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Comment by pmd1nh** [comment]
added a comment to test, no need for new PB.

File: dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/test-applications/GetMappingAsyncDispatchTest.war/src/servlets/ServletMapping.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            sBuilderResponse.append(" --]]] Result [PASS]");
+        else
+            sBuilderResponse.append(" --]]] Result [FAIL]");
+
+
+        //Client check this header.
+        response.setHeader("TestResult", sBuilderResponse.toString());  
+        LOG.info("Test ServletMapping dispatched Done.");
+    }
+}
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/test-applications/GetMappingAsyncDispatchTest.war/src/servlets/ServletMapping.java]: Just add a new line here

**Comment by pmd1nh** [comment]
#libby

**Comment by pmd1nh** [comment]
rebased and updated comments.

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

**Review by pnicolucci — APPROVED** [review]

**Comment by pmd1nh** [comment]
#libby

**Pull request merged**